cloud/scope/machine.go

```diff
@@ -192,6 +192,57 @@ func (m *MachineScope) ensureInstanceUnique(instanceName string) (*vpcv1.Instanc
 	return nil, nil
 }
 
+// CreateVPCLoadBalancerPoolMember creates a new pool member and adds it to the load balancer pool.
+func (m *MachineScope) CreateVPCLoadBalancerPoolMember(internalIP *string, targetPort int64) (*vpcv1.LoadBalancerPoolMember, error) {
+	loadBalancer, _, err := m.IBMVPCClient.GetLoadBalancer(&vpcv1.GetLoadBalancerOptions{
+		ID: m.IBMVPCCluster.Status.VPCEndpoint.LBID,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	options := &vpcv1.CreateLoadBalancerPoolMemberOptions{}
+	options.SetLoadBalancerID(*m.IBMVPCCluster.Status.VPCEndpoint.LBID)
+	options.SetPoolID(*loadBalancer.Pools[0].ID)
+	options.SetTarget(&vpcv1.LoadBalancerPoolMemberTargetPrototype{
+		Address: internalIP,
+	})
+	options.SetPort(targetPort)
+
+	listOptions := &vpcv1.ListLoadBalancerPoolMembersOptions{}
+	listOptions.SetLoadBalancerID(*m.IBMVPCCluster.Status.VPCEndpoint.LBID)
+	listOptions.SetPoolID(*loadBalancer.Pools[0].ID)
+	listLoadBalancerPoolMembers, _, err := m.IBMVPCClient.ListLoadBalancerPoolMembers(listOptions)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to bind ListLoadBalancerPoolMembers to control plane %s/%s", m.IBMVPCMachine.Namespace, m.IBMVPCMachine.Name)
+	}
+
+	var ports []int64
+	if len(listLoadBalancerPoolMembers.Members) > 0 {
+		for i := 0; i <= len(listLoadBalancerPoolMembers.Members)-1; i++ {
+			ports = append(ports, *listLoadBalancerPoolMembers.Members[i].Port)
+		}
+	}
+
+	if *loadBalancer.ProvisioningStatus == string(infrav1beta1.VPCLoadBalancerStateActive) && !contains(ports, targetPort) && *internalIP != "0.0.0.0" {
```

Comment:
and why are we checking for `0.0.0.0`? is there a chance that IP address of instance will be that value?

Reply:
Yes, have noticed above IP being added with Ports due to slowness in fetching actual IP (first-time), so need above check to avoid adding incorrect entry.

Reply:
I feel we should avoid calling this function in first place if that's the ip address we get for the instance..

Reply:
Fixed.